src/engraving/rendering/score/lyricslayout.cpp

```diff
@@ -451,8 +451,11 @@ void LyricsLayout::createOrRemoveLyricsLine(Lyrics* item, LayoutContext& ctx)
         EngravingItem* endSegmentElement = endSegment->element(track);
         if (endSegment->tick() == endTick && endSegmentElement && endSegmentElement->type() == ElementType::CHORD) {
             // everything is OK if we have reached a chord at right tick on right track
-            // advance to next CR, or last segment if no next CR
-            endSegment = endSegment->nextCR(track, false);
+            // advance to next CR after duration of note, or last segment if no next CR
+            const Score* score = item->score();
+            const Chord* endChord = toChord(endSegmentElement);
+
+            endSegment = score ? score->tick2segment(endSegment->tick() + endChord->ticks(), true, SegmentType::ChordRest) : nullptr;
```

Comment:
I'm not a huge fan of tick2segment. It seems incredibly inefficient since it iterates over all measures in the score. However, I don't know what else we can use here

Reply:
@miiizen I agree with Roman, tick2segment is overkill in this case. I am looking at the implementation of `Segment::nextCR`, and it seems to me that if the first call returns null, you could simply call it again with `endSegment->nextCR(track, true)`, and it will give you what you need (i.e. the first CR segment with a CR in any voice), if I understand correctly?

Reply:
Another issue then comes from having intermediary chords in a different voice - for a melisma in voice 2 in this example `endSegment` starts at 2/4.  We need it to end up at 4/4.  Calling it the second time gives us the segment at 3/4, as this has a voice 1 chord.  The solution below finds the correct segment and behaves as we want at the end of a score.
<img width="280" alt="Screenshot 2024-09-12 at 13 09 52" src="https://github.com/user-attachments/assets/a95c0575-e7e9-46da-8965-5b2a392effae">
```suggestion
            const Segment* endChordSeg = endSegment;
            const Chord* endChord = toChord(endSegmentElement);

            endSegment = endChordSeg->nextCR(track, false);

            if (!endSegment) {
                endSegment = endChordSeg;
                while (endSegment && endSegment->tick() < endChord->tick() + endChord->ticks()) {
                    endSegment = endSegment->nextCR(muse::nidx, true);
                }
            }
```

Reply:
Ah you're right. Yes this look good to me now